net/netcheck/netcheck.go

```diff
@@ -980,6 +1023,45 @@ func (c *Client) finishAndStoreReport(rs *reportState, dm *tailcfg.DERPMap) *Rep
 	return report
 }
 
+var generate204URL = envknob.StringOrDefault(
+	"TS_NETCHECK_GENERATE_204_URL",
+	"http://controlplane.tailscale.com/generate_204",
+)
+
+var noRedirectClient = &http.Client{
+	// No redirects allowed
+	CheckRedirect: func(req *http.Request, via []*http.Request) error {
+		return http.ErrUseLastResponse
+	},
+
+	// Remaining fields are the same as the default client.
+	Transport: http.DefaultClient.Transport,
+	Jar:       http.DefaultClient.Jar,
+	Timeout:   http.DefaultClient.Timeout,
+}
+
+// checkCaptivePortal reports whether or not we think the system is behind a
+// captive portal, detected by making a request to a URL that we know should
+// return a "204 No Content" response and checking if that's what we get.
+//
+// The boolean return is whether we think we have a captive portal.
+func (c *Client) checkCaptivePortal(ctx context.Context) (bool, error) {
+	defer noRedirectClient.CloseIdleConnections()
+	ctx, cancel := context.WithTimeout(ctx, captivePortalTimeout)
```

Comment:
this would usually fail on most flights nowadays with how many RTTs there are involved

(at least it's not TLS too!)

Reply:
Hm, good point; lemme just rely on the `overallProbeTimeout` value instead of a sub-timeout here.